src/clojure/nrepl/tls_client_proxy.clj

```diff
@@ -0,0 +1,235 @@
+(ns nrepl.tls-client-proxy
+  (:require [nrepl.tls :as tls]
+            [nrepl.socket :as socket]
+            [nrepl.cmdline :as cmdline])
+  (:import (java.net Socket ServerSocket)
+           (java.io IOException BufferedInputStream BufferedOutputStream Closeable InputStream OutputStream)
+           (java.time.format DateTimeFormatter)
+           (java.time ZonedDateTime)))
+
+(def ^DateTimeFormatter pattern (DateTimeFormatter/ofPattern "yyyy-MM-dd HH:mm:ss z"))
+
+(defonce lock (Object.))
+
+(defn atomic-println [& args]
```

Comment:
Please make sure that all public functions have docstrings. If something's not part of the public API then mark it as private explicitly.